src/plugins/mongodb/mongodb_client_config.cpp

```diff
@@ -201,35 +141,12 @@ MongoDBClientConfig::log(Logger *logger, const char *component, const char *inde
 	switch (mode_) {
 	case REPLICA_SET: {
 		logger->log_info(component, "%smode:   replica set", indent);
-		logger->log_info(component, "%shosts:", indent);
-		std::vector<mongo::HostAndPort>::iterator i;
-		for (i = replicaset_hostports_.begin(); i != replicaset_hostports_.end(); ++i) {
-			logger->log_info(component, "%s  - %s:", indent, i->toString().c_str());
-		}
-
-		if (!auth_infos_.empty()) {
-			logger->log_info(component, "%sauth infos:", indent);
-			std::list<AuthInfo>::iterator a;
-			for (a = auth_infos_.begin(); a != auth_infos_.end(); ++a) {
-				logger->log_info(
-				  component, "%s  - %s @ %s", indent, a->username.c_str(), a->dbname.c_str());
-			}
-		}
-	} break;
-
+	}
 	default: {
 		logger->log_info(component, "%smode:   connection", indent);
-		logger->log_info(component, "%shost:   %s", indent, conn_hostport_.toString().c_str());
-		if (!auth_infos_.empty()) {
-			logger->log_info(component, "%sauth infos:", indent);
-			std::list<AuthInfo>::iterator a;
-			for (a = auth_infos_.begin(); a != auth_infos_.end(); ++a) {
-				logger->log_info(
-				  component, "%s  - %s @ %s", indent, a->username.c_str(), a->dbname.c_str());
-			}
-		}
 	} break;
 	}
+	logger->log_info(component, "%suri:    %s", indent, conn_uri_.to_string().c_str());
```

Comment:
This may spill authentication data into the log. I think we should avoid that and at blank out the password.

Reply:
Fixed!